apps/dav/lib/BulkUpload/MultipartRequestParser.php

```
@@ -217,10 +217,6 @@ private function readPartContent(int $length, string $md5): string {
 			throw new Exception("Fail to read part's content.");
 		}
 
-		if (feof($this->stream)) {
-			throw new Exception("Unexpected EOF while reading stream.");
-		}
-
```

Was this causing issues?

---

yes
if the file has size 0 then upon receiving the part, `feof($this->stream)` will be true immediately after finding the boundaries of the part and this will throw instead of saving the empty file

---

Can we change to:


```suggestion
		if ($length !== 0 && feof($this->stream)) {
			throw new Exception("Unexpected EOF while reading stream.");
		}

```

---

done